ui/app/templates/components/role-aws-edit.hbs

```diff
@@ -21,30 +21,41 @@
   </p.levelLeft>
 </PageHeader>
 
-<Toolbar>
-  <ToolbarActions>
-    {{#if (eq mode "show")}}
-      {{#if (or model.canUpdate model.canDelete)}}
+{{#if (eq mode "show")}}
+  <Toolbar>
+    <ToolbarActions>
+      {{#if model.canGenerate}}
         <ToolbarSecretLink
           @secret={{model.id}}
-          @mode="edit"
-          @replace=true
+          @mode="credentials"
+          @data-test-backend-credentials="iam"
         >
-          Edit role
+          Generate credentials
         </ToolbarSecretLink>
       {{/if}}
-      {{#if model.canGenerate}}
+      {{#if (and model.canGenerate (or model.canDelete model.canUpdate))}}
+        <div class="toolbar-separator" />
+      {{/if}}
+      {{#if model.canDelete}}
+        <ConfirmAction
+          @buttonClasses="toolbar-link"
+          @onConfirmAction={{action "delete"}}
+        >
+          Delete role
+        </ConfirmAction>
+      {{/if}}
+      {{#if (or model.canUpdate model.canDelete)}}
```

Comment:
Any idea why the Edit button isn't showing up with just `{{#if model.canUpdate}}`?

Reply:
Hmm it should - I think previously it had this because delete was on the edit page. I can take a look.

Reply:
Ok, it's because the model it's a capabilities object, and the actual attribute is `canEdit`, so just `#if model.canEdit` should work here - looks like that's true of other secret models too.

https://github.com/hashicorp/vault/blob/d7e441a78a2743adb8587f7261dd82f877f3a5c7/ui/app/models/role-aws.js#L67